test/testdata/rewriter/concern.rb.rewrite-tree.exp

```diff
@@ -0,0 +1,192 @@
+class <emptyTree><<C <root>>> < (::<todo sym>)
+  module <emptyTree>::<C ActiveSupport>::<C Concern><<C <todo sym>>> < ()
+  end
+
+  module <emptyTree>::<C Foo><<C <todo sym>>> < ()
+    def instance_method<<C <todo sym>>>(&<blk>)
+      <emptyTree>
+    end
+
+    module <emptyTree>::<C ClassMethods><<C <todo sym>>> < ()
+      def a_class_method<<C <todo sym>>>(&<blk>)
+        <emptyTree>
+      end
+
+      ::Sorbet::Private::Static.keep_def(<self>, :a_class_method)
+    end
+
+    <self>.extend(<emptyTree>::<C ActiveSupport>::<C Concern>)
+
+    <self>.extend(::T::Helpers)
```

Comment:
This is going to silence some exceptions that happen at runtime.

```ruby
module Foo
  extend ActiveSupport::Concern
  class_methods do
    def a_class_method
    end
  end

  abstract! # exception at runtime, no static error
end
```

I'm not sure of a way to handle that.

Can you add this as a test, and confirm that this is a tradeoff that you're ok with? Stripe's codebase will not be using this so I'm fine with this as long as you are.

Reply:
@jez Thank you so much for bringing this up, I missed it. We don't want to make this tradeoff and happy to implement a solution as part of this PR.

We had internal discussions and we thought of adding a `T::Helpers::WithoutRuntime.mixes_in_class_methods` (similar to `T::Sig::WithoutRuntime.sig`) which has no runtime implementation (since `ActiveSupport::Concern` does its own implementation at runtime anyway).

This way instead of inserting a `extend(T::Helpers)` we can explicitly call `T::Helpers::WithoutRuntime.mixes_in_class_methods`. Let me know what do you think 😊

Reply:
Sounds good. I'm leaning towards something like `<Magic>.<mixes_in_class_methods>` because I don't know whether we want to implement this at runtime, and using `<Magic>` would make it impossible to be used outside of Sorbet internals.

If we do want to implement this at runtime, it would have to accept `self` as an argument, and then we may as well make it a class method on `T::Helpers` because it works at runtime.

Regardless, Stripe is having a global remote offsite this week, so I'll have to get back to you in a few days after my team has a chance to meet and discuss this.

Reply:
@jez I started implementing this and I'm at a crossroads while trying to satisfy typechecking. I initially opted for adding a simple `Magic` class that declares `mixes_in_class_methods`in the rewriter stage. However, I'm having trouble generating the method on the **singleton class** of `Magic`.

I'm curious if that's the route you wanted to take or should I instead follow a similar approach to https://github.com/sorbet/sorbet/pull/2194/files for typechecking errors related to `Magic#mixes_in_class_methods`.

Reply:
Ended up defining the method in `GlobalState.cc` similar to other `<Magic>` usages and it's simpler now.

Reply:
> Sounds good. I'm leaning towards something like `<Magic>.<mixes_in_class_methods>` because I don't know whether we want to implement this at runtime, and using `<Magic>` would make it impossible to be used outside of Sorbet internals.
> 
> If we do want to implement this at runtime, it would have to accept `self` as an argument, and then we may as well make it a class method on `T::Helpers` because it works at runtime.

@KaanOzkan Could you make the new Magic mixes_in_class_methods method take `self` as an argument?